eng/pipelines/merge-docs-main-to-live.yml

```diff
@@ -0,0 +1,95 @@
+parameters:
+- name: WhatIfPreference
+  type: boolean
+  default: false
+
+- name: DocsRepos
+  type: object
+  default:
+    - RepoName: azure-docs-sdk-java
+      RepoOwner: Azure
+      SafeName: Java
+      AllowlistPath: docsms-allowlist/java-allowlist.txt
+    - RepoName: azure-docs-sdk-python
+      RepoOwner: MicrosoftDocs
+      SafeName: Python
+      AllowlistPath: docsms-allowlist/python-allowlist.txt
+    - RepoName: azure-docs-sdk-node
+      RepoOwner: MicrosoftDocs
+      SafeName: JS
+      AllowlistPath: docsms-allowlist/js-allowlist.txt
+
+- name: TargetBranch
+  type: string
+  default: live
+
+jobs:
+- ${{ each repo in parameters.DocsRepos }}:
+  - job: 'Release${{repo.SafeName}}DocsMs'
+    pool:
+      name: azsdk-pool-mms-ubuntu-2004-general
+      vmImage: MMSUbuntu20.04
+    variables:
+      DocRepoLocation: $(Pipeline.Workspace)/docs
+    steps:
+      # Checkout tool repo
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+      # Checkout docs repo
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+        parameters:
+          SkipDefaultCheckout: true
+          Repositories:
+          - Name: ${{ repo.RepoOwner }}/${{ repo.RepoName }}
+            WorkingDirectory: $(DocRepoLocation)/${{repo.RepoName}}
+          Paths:
+          - /*
+
+      - task: Powershell@2
+        inputs:
+          pwsh: true
+          filePath: eng/scripts/create-tag-for-current-branch.ps1
+          arguments: 
+            -GithubUrl "https://$(azuresdk-github-pat)@github.com/${{repo.RepoOwner}}/${{repo.RepoName}}.git" `
+            -WhatIf:$${{ parameters.WhatIfPreference }}
+          workingDirectory: $(DocRepoLocation)/${{repo.RepoName}}
+        displayName: Create tag for base branch
+
+      - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+        parameters:
+          WorkingDirectory: $(DocRepoLocation)/${{repo.RepoName}}
+          DefaultBranchVariableName: ${{repo.RepoName}}
```

Comment:
Why are we using the RepoName as the name of the variable for the default branch? Do we expect this to be set mutiple times?

Reply:
tool repo and docs repos are having default branches on its own. (Though they are all using 'main')
We need to use different var name for different repo for accuracy.

Reply:
```suggestion
          DefaultBranchVariableName: '${{repo.RepoName}}-DefaultBranch'
```

Ok just to help clarify this perhaps we should add "-DefaultBranch" to the name.